test/runner/e2e_delete_rel_test.cpp

```diff
@@ -19,104 +19,316 @@ class DeleteRelTest : public DBTest {
+        // scans the backward list of the knows table. If this test fails (eg. we have updated the
+        // planner), we should consider changing to a new query which scans the BWD list with
+        // the new planner.
+        validateQueryBestPlanJoinOrder(query, "HJ(a){E(a)S(b)}{S(a)}");
+        conn->beginWriteTransaction();
+        // We insert 10 rels: person[0..10]->animal0.
+        for (auto i = 0u; i < 10; i++) {
+            ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("person", "animal", i, 0 /* dstID */))
+                            ->isSuccess());
+        }
+        // Then we delete rels: person[2,4,6,8]->animal0.
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 2, 0 /* dstID */))->isSuccess());
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 4, 0 /* dstID */))->isSuccess());
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 6, 0 /* dstID */))->isSuccess());
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 8, 0 /* dstID */))->isSuccess());
+        commitOrRollbackConnectionAndInitDBIfNecessary(isCommit, transactionTestType);
+        vector<string> expectedResult =
+            isCommit ? vector<string>{"0", "1", "3", "5", "7", "9"} : vector<string>{};
+        ASSERT_EQ(
+            TestHelper::convertResultToString(*conn->query(query), true /* checkOutputOrder */),
+            expectedResult);
+    }
 
-    auto actualResult = TestHelper::convertResultToString(
-        *conn->query("MATCH (:animal)-[e:knows]->(:person) return e.length"),
-        true /* checkOutputOrder */);
-    // After insertion and deletion, we are expected to see only one new rel: animal52->person0.
-    vector<string> expectedResult;
-    for (auto i = 0; i < NUM_RELS_FOR_ANIMAL_KNOWS_PERSON; i++) {
-        expectedResult.push_back(to_string(i));
+    void deleteMultipleRels(bool isCommit, TransactionTestType transactionTestType) {
+        conn->beginWriteTransaction();
+        // Delete all rels from animal[10-49]->person0.
+        ASSERT_TRUE(conn->query("MATCH (a:animal)-[e:knows]->(p:person) WHERE a.ID >= 10 AND "
+                                "a.ID <= 49 AND p.ID = 0 DELETE e;")
+                        ->isSuccess());
+        commitOrRollbackConnectionAndInitDBIfNecessary(isCommit, transactionTestType);
+        vector<string> expectedResult;
+        if (isCommit) {
+            expectedResult = vector<string>{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "50"};
+        } else {
+            for (auto i = 0u; i <= 50; i++) {
+                expectedResult.push_back(to_string(i));
+            }
+        }
+        ASSERT_EQ(TestHelper::convertResultToString(
+                      *conn->query("MATCH (a:animal)-[e:knows]->(p:person) RETURN e.length"),
+                      true /* checkOutputOrder */),
```

Comment:
Why are you checking the output order in a query that does not have ORDER BY?

Reply:
The reason is that if the `checkOutputOrder` is false, the testHelper will sort the actual test results. 
Anyways, i changed checkOutputOrder to false, and sort the expectedResult when comparing with the actualResult.